OpenRA.Mods.Common/Widgets/Logic/Installation/DownloadPackagesLogic.cs

```diff
@@ -21,24 +20,30 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 {
 	public class DownloadPackagesLogic
 	{
+		static readonly string[] SizeSuffixes = { "bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB" };
 		readonly Widget panel;
+		readonly string modId;
 		readonly string mirrorListUrl;
 		readonly ProgressBarWidget progressBar;
 		readonly LabelWidget statusLabel;
 		readonly Action afterInstall;
 		string mirror;
-		static readonly string[] SizeSuffixes = { "bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB" };
 
 		[ObjectCreator.UseCtor]
-		public DownloadPackagesLogic(Widget widget, Action afterInstall, string mirrorListUrl)
+		public DownloadPackagesLogic(Widget widget, Action afterInstall, string mirrorListUrl, string modId)
 		{
 			this.mirrorListUrl = mirrorListUrl;
 			this.afterInstall = afterInstall;
+			this.modId = modId;
 
 			panel = widget.Get("INSTALL_DOWNLOAD_PANEL");
 			progressBar = panel.Get<ProgressBarWidget>("PROGRESS_BAR");
 			statusLabel = panel.Get<LabelWidget>("STATUS_LABEL");
 
+			var modName = ModMetadata.AllMods[modId].Title;
```

Comment:
You could improve separation by passing the modName (`modTitle`?) instead of the modId and doing this lookup.